frontend/app/src/components/staking/liquity/LiquityStatistics.vue

```diff
@@ -0,0 +1,348 @@
+  );
+
+  // B = Total Gains Stability Pool + Liquidations Gains (current price) + Rewards (current price)
+  const totalWithdrawals = statisticVal.totalUsdGainsStabilityPool
+    .plus(liquidationGainsInCurrentPrice)
+    .plus(rewardsInCurrentPrice);
+
+  // C = A - Current deposited amount
+  const diffDeposited = expectedAmount.minus(poolVal.deposited.amount);
+
+  // convert C to USD at current price
+  const diffDepositedInCurrentUsdPrice = diffDeposited.multipliedBy(
+    get(lusdPrice) ?? One
+  );
+
+  // PnL = B - C
+  return totalWithdrawals.minus(diffDepositedInCurrentUsdPrice);
+});
+
+const css = useCssModule();
+</script>
+
+<template>
+  <card :loading="loading">
+    <template #title>
+      {{ tc('liquity_statistic.title') }}
+    </template>
+    <template #details>
+      <v-btn-toggle v-model="current" dense mandatory>
+        <v-btn :value="true">
+          {{ tc('liquity_statistic.switch.current') }}
+        </v-btn>
+        <v-btn :value="false">
+          {{ tc('liquity_statistic.switch.historical') }}
+        </v-btn>
+      </v-btn-toggle>
+    </template>
+    <template v-if="statisticWithAdjustedPrice">
+      <v-row class="ma-n6" :class="css.large">
+        <v-col md="6" class="pa-6 py-8 d-flex justify-space-between">
+          <div>{{ tc('liquity_statistic.total_gains_stability_pool') }}</div>
+          <v-sheet>
+            <amount-display
+              :value="statisticWithAdjustedPrice.totalUsdGainsStabilityPool"
+              :fiat-currency="CURRENCY_USD"
+              class="font-weight-bold"
+            />
+          </v-sheet>
+        </v-col>
+        <v-col md="6" class="pa-6 py-8 d-flex justify-space-between">
+          <div>{{ tc('liquity_statistic.total_gains_staking') }}</div>
+          <v-sheet>
+            <amount-display
+              :value="statisticWithAdjustedPrice.totalUsdGainsStaking"
+              :fiat-currency="CURRENCY_USD"
+              class="font-weight-bold"
+            />
+          </v-sheet>
+        </v-col>
+      </v-row>
+
+      <v-expansion-panels multiple class="pt-4">
+        <v-expansion-panel elevation="0">
+          <v-expansion-panel-content>
+            <v-row class="ma-n6">
+              <v-col md="6" class="pa-6">
+                <div>
+                  <v-divider />
+                  <div class="text-right py-4">
+                    <div class="font-weight-medium pb-2" :class="css.label">
+                      {{
+                        tc('liquity_statistic.total_deposited_stability_pool')
+                      }}
+                    </div>
+                    <balance-display
+                      :asset="LUSD_ID"
+                      :value="{
+                        amount:
+                          statisticWithAdjustedPrice.totalDepositedStabilityPool,
+                        usdValue:
+                          statisticWithAdjustedPrice.totalDepositedStabilityPoolUsdValue
+                      }"
```

Comment:
any chance we could do a computed out of this? maybe even make it composable util. I think it might be useful in other places too.


```typescript
const exampleObject = {
  a: 1,
  b: 2
};

const useRefMap = <I extends object, O>(
  inp: I,
  map: (inp: I) => O
): ComputedRef<O> => computed(() => map(get(inp)));

const comp = useRefMap(exampleObject, x => ({
  c: x.a
}));
```